apps/dotcom/client/src/fairy/fairy-app/managers/FairyAppPersistenceManager.ts

```diff
@@ -159,6 +160,77 @@ export class FairyAppPersistenceManager extends BaseFairyAppManager {
 
 		// Store the flush function for cleanup
 		this.autoSaveCleanupFns.push(() => updateFairyState.flush())
+
+		// Start watching chat messages for append-only persistence
+		this.startChatMessageAppending(fileId)
+	}
+
+	/**
+	 * Start watching chat history and append complete messages to database.
+	 *
+	 * @param fileId - The file ID to append messages to
+	 */
+	private startChatMessageAppending(fileId: string) {
```

Comment:
A bunch of duplication with `startAutoSave`, but fine with this just not to break something.